test/jasmine/tests/splom_test.js

```diff
@@ -359,6 +359,36 @@ describe('Test splom trace defaults:', function() {
     });
 });
 
+describe('Test splom trace calc step:', function() {
+    var gd;
+
+    function _calc(opts, layout) {
+        gd = {};
+
+        gd.data = [Lib.extendFlat({type: 'splom'}, opts || {})];
+        gd.layout = layout || {};
+        supplyAllDefaults(gd);
+        Plots.doCalcdata(gd);
+    }
+
+    it('should skip dimensions with conflicting axis types', function() {
+        _calc({
+            dimensions: [{
+                values: [1, 2, 3]
+            }, {
+                values: [2, 1, 2]
+            }]
+        }, {
+            xaxis: {type: 'category'},
+            yaxis: {type: 'linear'}
+        });
+
+        var cd = gd.calcdata[0][0];
+
+        expect(cd.t._scene.matrixOptions.data).toBeCloseTo2DArray([[2, 1, 2]]);
```

Comment:
This is good, but I have various questions about what happens downstream from `calc` with the missing dimension(s)... like, `cdata` and `ldata` arrays don't match up with the dimension array anymore, does this get handled correctly in plotting? The next commit (`visibleDims`) seems like it assuages my concerns a bit, but would you mind including a mismatched-type dimension (somewhere in the middle of `dimensions`) in a mock so we can see it all the way to the end?

Reply:
Good call, done in -> https://github.com/plotly/plotly.js/pull/2899/commits/390f29202e4230035e094e9731f4f643e5ca9ea9?short_path=10d6861#diff-10d68616ed667ce29da903fcd8380c29